### PR TITLE
Social manager flat C APIs

### PR DESCRIPTION
**Pull request by jplafonta**
Adding social manager flat C APIs and updating sample to demonstrate their usage.

### CODE REVIEW

File: Build/Microsoft.Xbox.Services.140.UWP.Cpp/Microsoft.Xbox.Services.140.UWP.Cpp.vcxproj [hunk 1]
```diff
@@ -141,6 +141,7 @@
     <IntDir>$(ProjectDir)..\..\Obj\$(Configuration)\$(Platform)\$(ProjectFolder)\</IntDir>
   </PropertyGroup>
   <ItemGroup>
+    <ClCompile Include="..\..\Source\Services\Social\Manager\C\social_manager_c.cpp" />
```
Comment (rakhad) [inline Build/Microsoft.Xbox.Services.140.UWP.Cpp/Microsoft.Xbox.Services.140.UWP.Cpp.vcxproj]: It seems you deleted this file?

Comment (rakhad) [inline Build/Microsoft.Xbox.Services.140.UWP.Cpp/Microsoft.Xbox.Services.140.UWP.Cpp.vcxproj]: $(MSBuildThisFileDirectory)

File: Build/Microsoft.Xbox.Services.140.UWP.Cpp/Microsoft.Xbox.Services.140.UWP.Cpp.vcxproj.filters [hunk 2]
```diff
@@ -832,6 +826,9 @@
     <ClCompile Include="$(MSBuildThisFileDirectory)..\..\Source\System\xbox_live_mutex.cpp">
       <Filter>C++ Source\System</Filter>
     </ClCompile>
+    <ClCompile Include="..\..\Source\Services\Social\Manager\C\social_manager_c.cpp">
```
Comment (rakhad) [inline Build/Microsoft.Xbox.Services.140.UWP.Cpp/Microsoft.Xbox.Services.140.UWP.Cpp.vcxproj.filters]: $(MSBuildThisFileDirectory)

File: Include/xsapi-c/presence_c.h [hunk 3]
```diff
@@ -10,59 +10,129 @@
...
+    XBL_PRESENCE_TITLE_VIEW_STATE_FILLED,
+
+    /// <summary>The title's view is snapped with another application using a part of the screen.</summary>
+    XBL_PRESENCE_TITLE_VIEW_STATE_SNAPPED,
+
+    /// <summary>The title's running in the background and is not visible.</summary>
+    XBL_PRESENCE_TITLE_VIEW_STATE_BACKGROUND
+} XBL_PRESENCE_TITLE_VIEW_STATE;
+
+/// <summary> Defines values used to set the level of presence detail return from the service. Choosing proper detail level could help the performance of the API.</summary>
```
Comment (rakhad) [inline Include/xsapi-c/presence_c.h]: nit: break this into 2 lines

File: Include/xsapi-c/social_manager_c.h [hunk 4]
```diff
@@ -14,230 +14,561 @@ extern "C" {
...
+    /// <summary>Social graph removal complete</summary>
+    XBL_SOCIAL_EVENT_TYPE_LOCAL_USER_REMOVED,
+
+    /// <summary>Xbox Social User Group load complete (will only trigger for views that take a list of users)</summary>
+    XBL_SOCIAL_EVENT_TYPE_SOCIAL_USER_GROUP_LOADED,
+
+    /// <summary>Social user group updated</summary>
+    XBL_SOCIAL_EVENT_TYPE_SOCIAL_USER_GROUP_UPDATED,
+
+    /// <summary>unknown.</summary>
```
Comment (rakhad) [inline Include/xsapi-c/social_manager_c.h]: >unknown. [](start = 17, length = 8)

nit: to be consistent try Unknown

**Comment by rakhad** [comment]
> // Copyright (c) Microsoft Corporation

Thoughts: Is comment consistency important? I think it creates a nicer final product. We should agree on a set of style guidelines, like whether comments should end in periods.

---
Refers to: Include/xsapi-c/social_manager_c.h:1 in 0b1caae. [](commit_id = 0b1caae9f7d3f9f69e1a05da02e6d4148f525829, deletion_comment = False)

File: Include/xsapi-c/social_manager_c.h [hunk 5]
```diff
@@ -14,230 +14,561 @@ extern "C" {
...
-SocialManagerAddLocalUser(
-    _In_ XBL_XBOX_LIVE_USER *user,
-    _In_ XSAPI_SOCIAL_MANAGER_EXTRA_DETAIL_LEVEL extraLevelDetail,
-    _Out_ PCSTR* errMessage
-    );
-
+XblSocialUserGroupGetUsersTrackedByGroup(
+    _In_ XBL_XBOX_SOCIAL_USER_GROUP* group,
+    _Out_ XBL_XBOX_USER_ID_CONTAINER* trackedUsers
+    ) XBL_NOEXCEPT;
```
Comment (rakhad) [inline Include/xsapi-c/social_manager_c.h]: How would this work without a count param?

File: InProgressSamples/Social/UWP/C/SocialManagerIntegration.cpp [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+
+void Game::AddUserToSocialManager(XBL_XBOX_LIVE_USER* user)
+{
+    stringstream_t source;
+    source << _T("Adding user ");
+    source << user->gamertag;
+    source << _T(" to SocialManager");
+    Log(source.str());
+
+    XblSocialManagerAddLocalUser(user, XBL_SOCIAL_MANAGER_EXTRA_DETAIL_LEVEL_ALL);
```
Comment (rakhad) [inline InProgressSamples/Social/UWP/C/SocialManagerIntegration.cpp]: >XblSocialManagerAddLocalUser [](start = 4, length = 28)

nit: add logging to status

File: InProgressSamples/Social/UWP/C/SocialManagerIntegration.cpp [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+    {
+        xuidListArray[i] = xuidList[i].data();
+    }
+
+    for (auto group : m_socialGroups)
+    {
+        if (group->socialUserGroupType == XBL_SOCIAL_USER_GROUP_TYPE_USER_LIST_TYPE && group->localUser == user)
+        {
+            XblSocialManagerUpdateSocialUserGroup(group, xuidListArray, (uint32_t)xuidList.size());
+            return;
```
Comment (rakhad) [inline InProgressSamples/Social/UWP/C/SocialManagerIntegration.cpp]: >return [](start = 12, length = 6)

delete[] xuidListArray; before returning

File: Source/System/ppltasks_extra.h [hunk 8]
```diff
@@ -7,6 +7,7 @@
 #include <agents.h>
 #include <memory>
 #include <chrono>
+#include "Utils_WinRT.h"
```
Comment (rakhad) [inline Source/System/ppltasks_extra.h]: Why was this added?

File: Include/xsapi-c/social_manager_c.h [hunk 9]
```diff
@@ -14,230 +14,561 @@ extern "C" {
...
+    _Out_ XBL_XBOX_SOCIAL_USER_GROUP** group
+    ) XBL_NOEXCEPT;
+
+/// <summary>
+/// Constructs a social Xbox Social User Group, which is a collection of users with social information
+/// The result of a user group being loaded will be triggered through the XBL_SOCIAL_EVENT_TYPE_SOCIAL_USER_GROUP_LOADED event in XblSocialManagerDoWork
+/// </summary>
+/// <param name="user">Xbox Live User</param>
+/// <param name="xboxUserIdList">List of users to populate the Xbox Social User Group with. This is currently capped at 100 users total.</param>
+/// <param name="xboxUserIdList">The number of items in the xboxUserIdList.</param>
```
Comment (rakhad) [inline Include/xsapi-c/social_manager_c.h]: >xboxUserIdList [](start = 17, length = 14)

xboxUserIdListCount

File: Include/xsapi-c/social_manager_c.h [hunk 10]
```diff
@@ -14,230 +14,561 @@ extern "C" {
...
-    );
-
+    _Out_ XBL_XBOX_SOCIAL_USER_GROUP** group
+    ) XBL_NOEXCEPT;
+
+/// <summary>
+/// Destroys a created social Xbox Social User Group
+/// This will stop updating the Xbox Social User Group and remove tracking for any users the Xbox Social User Group holds
+/// </summary>
+/// <param name="socialUserGroup">The social Xbox Social User Group to destroy and stop tracking</param>
```
Comment (rakhad) [inline Include/xsapi-c/social_manager_c.h]: >socialUserGroup [](start = 17, length = 15)

group

File: Include/xsapi-c/social_manager_c.h [hunk 11]
```diff
@@ -14,230 +14,561 @@ extern "C" {
...
+    _Out_ XBL_XBOX_SOCIAL_USER_GROUP** loadedGroup
+    ) XBL_NOEXCEPT;
+
+/// <summary>
+/// Constructs a social Xbox Social User Group, which is a collection of users with social information
+/// The result of a user group being loaded will be triggered through the XBL_SOCIAL_EVENT_TYPE_SOCIAL_USER_GROUP_LOADED event in XblSocialManagerDoWork
+/// </summary>
+/// <param name="user">Xbox Live User</param>
+/// <param name="presenceDetailLevel">The restriction of users based on their presence and title activity</param>
+/// <param name="relationshipFilter">The restriction of users based on their relationship to the calling user</param>
```
Comment (rakhad) [inline Include/xsapi-c/social_manager_c.h]: >relationshipFilter [](start = 17, length = 18)

filter

File: Source/System/ppltasks_extra.h (hunk 8 above)
Comment (jplafonta) [reply Source/System/ppltasks_extra.h]: Unneeded, I changed the file and the reverted the change but forgot to remove this header.

---
In reply to: [173710245](https://github.com/Microsoft/xbox-live-api/pull/329#discussion_r173710245) [](ancestors = 173710245)

File: Build/Microsoft.Xbox.Services.140.UWP.Cpp/Microsoft.Xbox.Services.140.UWP.Cpp.vcxproj (hunk 1 above)
Comment (jplafonta) [reply Build/Microsoft.Xbox.Services.140.UWP.Cpp/Microsoft.Xbox.Services.140.UWP.Cpp.vcxproj]: will regen project files

---
In reply to: [173640613](https://github.com/Microsoft/xbox-live-api/pull/329#discussion_r173640613) [](ancestors = 173640613)

**Comment by jplafonta** [comment]
> // Copyright (c) Microsoft Corporation

I agree. I just copied these from the xsapi core types. We can go through and cleanup all the comments before we release if we really think its a problem, but I'm just going to leave for now

---
In reply to: [372083187](https://github.com/Microsoft/xbox-live-api/pull/329#issuecomment-372083187) [](ancestors = 372083187)

---
Refers to: Include/xsapi-c/social_manager_c.h:1 in 0b1caae. [](commit_id = 0b1caae9f7d3f9f69e1a05da02e6d4148f525829, deletion_comment = False)

File: InProgressSamples/Social/UWP/C/SocialManagerIntegration.cpp (hunk 6 above)
Comment (jplafonta) [reply InProgressSamples/Social/UWP/C/SocialManagerIntegration.cpp]: Not sure what you mean by this

---
In reply to: [173641162](https://github.com/Microsoft/xbox-live-api/pull/329#discussion_r173641162) [](ancestors = 173641162)

File: InProgressSamples/Social/UWP/C/SocialManagerIntegration.cpp (hunk 7 above)
Comment (jplafonta) [reply InProgressSamples/Social/UWP/C/SocialManagerIntegration.cpp]: good catch, added

---
In reply to: [173641217](https://github.com/Microsoft/xbox-live-api/pull/329#discussion_r173641217) [](ancestors = 173641217)
